src/python/controller/extract/extract.py

```
@@ -117,8 +165,25 @@ def extract_archive(archive_path: str, out_dir_path: str):
             # Try to create the outdir path
             if not os.path.exists(out_dir_path):
                 os.makedirs(out_dir_path)
+
+            # Pre-validate member paths for zip/tar before extraction
+            pre_validated = Extract._pre_validate_members(archive_path, out_dir_path)
+
             patoolib.extract_archive(archive_path, outdir=out_dir_path, interactive=False)
         except FileNotFoundError as e:
             raise ExtractError(str(e))
```

Ruff (0.15.2)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)